File: Programming Basics with JS/T07 - Nested Loops/exercise/04. Train the Trainers.js
function trainTrainers(input) {
    let index = 0;
    let judges = Number(input[index]);
    index++;

    let command = input[index];
    let sumGrades = 0;
    let themes = 0;

    while (command !== 'Finish') {
        themes++;
        let name = command;
        let tempSumGrade = 0;

        for (let i = 0; i < judges; i++) {
            index++;
            let grade = Number(input[index]);
            tempSumGrade += grade;
        }

        let tempAvgGrade = tempSumGrade / judges;
        sumGrades += tempAvgGrade;
        console.log(`${name} - ${tempAvgGrade.toFixed(2)}.`);
        index++;
        command = input[index];
    }

    let avgGrade = sumGrades / themes;
    console.log(`Student's final assessment is ${avgGrade.toFixed(2)}.`);
}
trainTrainers(["2", "While-Loop", "6.00", "5.50", "For-Loop", "5.84", "5.66", "Finish"]);

File: Programming Basics with Python/T07 - Nested Loops/exercise/04. Train The Trainers.py
people = int(input())
command = input()
sum_grade = 0
counter = 0

while command != "Finish":
    name = command
    temporarily_sum_grade = 0
    counter += 1

    for i in range (0, people):
        temporarily_grade = float(input())
        temporarily_sum_grade += temporarily_grade

    tempAvgGrade = temporarily_sum_grade / people
    sum_grade += tempAvgGrade
    print(f"{name} - {tempAvgGrade:.2f}.")
    command = input()

average_grade = sum_grade / counter
print(f"Student's final assessment is {average_grade:.2f}.")